### PR TITLE
Blog app: Validations and model specs

### PR DESCRIPTION
**Pull request by BB-Simon**
**The Blog app** will be a classic example of a blog website which will be a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.

### In this PR:
- [x] Add the following validations:
    - For the User model:
      - Name must not be blank.
      - PostsCounter must be an integer greater than or equal to zero.
    - For the Post mode
      - Title must not be blank.
      - Title must not exceed 250 characters.
      - CommentsCounter must be an integer greater than or equal to zero.
      - LikesCounter must be an integer greater than or equal to zero.

- [x] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe User, type: :model do
+  subject { User.new(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.') }
+
+  before { subject.save }
```
Comment (TadesseDev) [inline spec/models/user_spec.rb]: - [x] Dear @BB-Simon, Your tests are great, But you are missing the tests for the `Like` and `Comment` models. Please remember we are expected to add unit tests for `User` ✅, `Post` ✅, `Comment` ❌, and `Like` ❌ models. Please don't forget to test all the custom methods in those models too.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @BB-Simon 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 1 above)
Comment (BB-Simon) [reply spec/models/user_spec.rb]: Hi @TadesseDev 
Thanks for your review, I have made the required changes :)

File: spec/models/comment_spec.rb
Comment (KDlamini) [inline spec/models/comment_spec.rb:1]: Well done adding the comment and likes specs as requested by the previous reviewer!!! 👏🏾💯 

However, some of the tests fail. in the snapshot below 7 out of the 19 tests fails. Kindly look into it and resolve them.

image

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @BB-Simon 🙋‍♀️,  

You have done an incredible job implementing some of the changes requested by the previous reviewer. 👏🏾👏🏾👏🏾  
However, there are still some changes that aren't addressed yet. You are almost there! 💪🏾 

## Highlights 🌞 

✔️ Good PR title and description
✔️ Tests for user, post, comment, and likes validations are well written
✔️ Well documented and professional README

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb
Comment (BB-Simon) [reply spec/models/comment_spec.rb:1]: Hi @KDlamini 
Thanks for reviewing now all tests are passing:)

spec

**Review by afanimartin — APPROVED** [review]
Hi @BB-Simon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Changes made as requested

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**